plugin/evm/orderbook/order_types.go

```diff
@@ -0,0 +1,107 @@
+package orderbook
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type ContractOrder interface {
+	EncodeToABI() ([]byte, error)
+	DecodeFromRawOrder(rawOrder interface{})
+}
+
+// LimitOrder type is copy of LimitOrder struct defined in Orderbook contract
+type LimitOrder struct {
+	AmmIndex          *big.Int       `json:"ammIndex"`
+	Trader            common.Address `json:"trader"`
+	BaseAssetQuantity *big.Int       `json:"baseAssetQuantity"`
+	Price             *big.Int       `json:"price"`
+	Salt              *big.Int       `json:"salt"`
+	ReduceOnly        bool           `json:"reduceOnly"`
+}
+
+// IOCOrder type is copy of IOCOrder struct defined in Orderbook contract
+type IOCOrder struct {
```

Comment:
do you think it might be a good idea to inherit from `LimitOrder`?

Reply:
No, I feel it's better to keep them separate, even if it means repeating the fields. Also, Go doesn't have regular inheritance, we could have LimitOrder as a field of IOCOrder, but it makes LimitOrder's fields nested fields. This could make serialisation/deserialisation complicated. Even though they are nested, go allows us to access it in both ways:

```
iocOrder.AmmIndex
```

```
iocOrder.LimitOrder.AmmIndex
```
both are valid